src/main/resources/compendium.yaml

```
@@ -0,0 +1,250 @@
+      urlPrefix:
+        default: v0
+tags:
+  - name: health
+    description: health ping
+  - name: protocol
+    description: Operations on protocols
+    externalDocs:
+      description: Learn more
+      url: https://github.com/higherkindness/compendium
+paths:
+  "/health":
+    get:
+      servers:
+        - url: '{protocol}://{host}:{port}'
+          description: Override base path for health response endpoint
+          variables:
+            protocol:
+              default: http
+            host:
+              default: localhost
+            port:
+              default: "8080"
+      tags:
+        - health
+      summary: health response endpoint
+      responses:
+        "200":
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/HealthResponse"
+          description: Service is healthy
+        "400":
+          description: Bad request
+        "404":
+          description: Not found
+        "500":
+          description: Internal server error
+  "/protocol/{protocolId}":
+    post:
+      tags:
+        - protocol
+      summary: store a protocol
+      parameters:
+        - name: protocolId
+          in: path
+          required: true
+          description: the id of the protocol - can be a string of any alphanumeric
+            character, dash (-), or period (.) with a length less than or equal to 255
+            characters
+          schema:
+            $ref: "#/components/schemas/ProtocolId"
+        - name: idlName
+          in: query
+          required: true
+          description: the name of the type of protocol being saved - must be one of avro,
+            protobuf, mu, openapi, or scala
+          schema:
+            $ref: "#/components/schemas/IdlName"
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/Protocol"
+        description: the protocol to be saved
+        required: true
+      responses:
+        "201":
+          description: successfully saved the protocol
+          headers:
+            Location:
+              schema:
+                type: string
+              description: location of the uploaded protocol
+          content:
+            application/json:
+              schema:
+                description: an integer representing the protocol version
+                type: integer
+        "400":
+          description: Bad request
+        "404":
+          description: Not found
+        "500":
+          description: Internal server error
+    get:
+      tags:
+        - protocol
+      summary: get a stored protocol by id
+      parameters:
+        - name: protocolId
+          in: path
+          required: true
+          description: the id of the protocol - can be a string of any alphanumeric
+            character, dash (-), or period (.) with a length less than or equal to 255
+            characters
```

It should not be necessary to describe it since there is a regex expression checking that in the `ProcotolId` type :wink: 

```suggestion
          description: the id of the protocol
```